### PR TITLE
feat: add experimental atlas to pull_translations

### PR DESCRIPTION
**Pull request by OmarIthawi**
### Description

This is an experimental off-by-defualt feature for moving the translation files ouside the repos.

Run `OPENEDX_ATLAS_PULL=true make translations` to use `atlas` to pull translations instead of `transifex`.

### Changes

 - Centralize i18n messages import in `src/i18n/index.js`
 - Add new `pull_translations` implementation via `atlas`
 - Use the new `i18n-imports.js` script from https://github.com/openedx/frontend-platform/pull/463

### Reviewers
 - [x] Get initial feedback (done by Brian)
 - [x] Get the dependencies bumped to bring up `intl-imports.js` script:
    - https://github.com/openedx/frontend-component-footer/pull/286
    - https://github.com/openedx/frontend-component-header/pull/329
    - https://github.com/openedx/frontend-platform/pull/463 
 - [x] Upgrade `frontend-platform@3.x.x` in exams library (blocker): https://github.com/openedx/frontend-lib-special-exams/pull/89
 - [x] Quick QA: <details><summary>Screenshot</summary>image</details> 
 - [x] Fix tests

### Sample expected outcome

The file tree after running the new `pull_translations` will look like the following:

All messages will be pulled fresh, without depending on any messages from the MFE or its depedencies.

```
frontend-app-learning $ tree src/i18n/
src/i18n/
├── index.js
└── messages
    ├── frontend-app-learning
    │   ├── ar.json
    │   ├── de.json
    │   ├── es_419.json
    │   ├── fr_CA.json
    │   ├── index.js
    │   ├── ru.json
    │   ├── uk.json
    │   └── zh_CN.json
    ├── frontend-component-footer
    │   ├── ar.json
    │   ├── de_DE.json
    │   ├── de.json
    │   ├── es_419.json
    │   ├── fr_CA.json
    │   ├── fr.json
    │   ├── hi.json
    │   ├── it_IT.json
    │   ├── it.json
    │   ├── pt.json
    │   ├── pt_PT.json
    │   ├── ru.json
    │   ├── uk.json
    │   └── zh_CN.json
    └── frontend-component-header
```

### Sample package `index.js` file

For each directory/package an `index.js` file will be generated if and only if there are messages in the files:

```js
// This file is generated by the Open edX generate-mfe-i18n-imports.
// Refer to docs yada yada ...
//

import messages0ar from './ar.json';
import messages1de from './de.json';
import messages2es_419 from './es_419.json';
import messages3fr_CA from './fr_CA.json';
import messages4ru from './ru.json';
import messages5uk from './uk.json';
import messages6zh_CN from './zh_CN.json';
const messages = {
  'ar': messages0ar,
  'zh-cn': messages6zh_CN,
  'uk': messages5uk,
  'ru': messages4ru,
  'fr-ca': messages3fr_CA,
  'es-419': messages2es_419,
  'de': messages1de,
};
export default messages;
```

### Sample MFE `src/i18n/index.js` file

A new `index.js` is going to be generated containing only downloaded languages.

```js
// This file is generated by the Open edX frontend-platform's i18n-imports commands.
//

import messages1frontendcomponentheader from './messages/frontend-component-header';
// footer is omitted due to not having translations
import messages2frontendapplearning from './messages/frontend-app-learning';

export default [
  messages1frontendcomponentheader,
  messages2frontendapplearning,
];
```

References
----------

This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Check the links above for full information about the overall project.

Internalization is being rearchitected in Open edX Python, XBlock, Micro-frontend, and other projects. There are a number of immediately visible changes:
 - Remove source and language translations from the repositories, hence no `.json`, `.po` or `.mo` files will be committed into the repos.
 - Add standardized `make extract_translations` in all repositories
 - Push user-facing messages strings into [openedx/openedx-translations](https://github.com/openedx/openedx-translations/).
 - Integrate root repositories with [openedx/openedx-atlas](https://github.com/openedx/openedx-atlas/) to pull translations on build/deploy time

Breaking Changes
----------------

One of the primary goals of the project is to **avoid breaking changes**. If you noticed any suspicious code, please raise your concern. But before that, please know the strategy we're following to avoid breaking changes

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @OmarIthawi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by OmarIthawi** [comment]
Hi, @edx/engage-squad I'd love a quick sanity check before I invest more time in shaping up this work, remove temp. references and add documentation.

**Comment by OmarIthawi** [comment]
Hi, @edx/engage-squad I'd love a quick sanity check before I invest more time in shaping up this work, remove temp. references and add documentation.

cc: @mphilbrick211, it turns out I can't tag the right team.

File: Makefile [hunk 1]
```diff
@@ -42,9 +43,22 @@ push_translations:
...
+ifeq ($(OPENEDX_ATLAS_PULL),)
 # Pulls translations from Transifex.
 pull_translations:
 	tx pull -f --mode reviewed --languages=$(transifex_langs)
+else
+# Experimental: OEP-58 Pulls translations using atlas
+pull_translations:
+	rm -rf src/i18n/messages
+	mkdir src/i18n/messages
+	cd src/i18n/messages && mkdir frontend-app-learning && cd frontend-app-learning && atlas pull --repository=Zeit-Labs/openedx-translations --branch=learning-mfe-translations --directory=translations/frontend-app-learning/src/i18n/messages
```
Comment (brian-smith-tcril) [inline Makefile]: just a note to make sure this gets switched back to the `openedx/openedx-translations` repo before this gets merged

**Review by brian-smith-tcril — COMMENTED** [review]
This is looking really great! Definitely a good pattern when combined with https://github.com/openedx/frontend-platform/pull/463!

Just one little note as a reminder so we don't forget and merge this while pointing to the Zeit Labs fork, but other than that this is looking good to go!

**Comment by mphilbrick211** [comment]
> Hi, @edx/engage-squad I'd love a quick sanity check before I invest more time in shaping up this work, remove temp. references and add documentation.
> 
> cc: @mphilbrick211, it turns out I can't tag the right team.

@OmarIthawi sorry for the late reply. @openedx/engage-squad - could someone please take a look at this? Thanks!

**Comment by itsjeyd** [comment]
Hey @OmarIthawi, just checking in to see if @brian-smith-tcril's feedback addressed your need for [a quick sanity check](https://github.com/openedx/frontend-app-learning/pull/1093#issuecomment-1492223446)? Or are you still looking for additional input from the engage-squad before continuing?

**Comment by OmarIthawi** [comment]
> Hey @OmarIthawi, just checking in to see if @brian-smith-tcril's feedback addressed your need for [a quick sanity check](https://github.com/openedx/frontend-app-learning/pull/1093#issuecomment-1492223446)? Or are you still looking for additional input from the engage-squad before continuing?

@itsjeyd I've got the initial review. Thanks for checking.

I'm now blocked by other PRs which brings the required `intl-imports.js` command:

  - https://github.com/openedx/frontend-component-footer/pull/286
  - https://github.com/openedx/frontend-component-header/pull/329
  - https://github.com/openedx/frontend-platform/pull/463

**Comment by brian-smith-tcril** [comment]
> > Hey @OmarIthawi, just checking in to see if @brian-smith-tcril's feedback addressed your need for [a quick sanity check](https://github.com/openedx/frontend-app-learning/pull/1093#issuecomment-1492223446)? Or are you still looking for additional input from the engage-squad before continuing?
> 
> @itsjeyd I've got the initial review. Thanks for checking.
> 
> I'm now blocked by other PRs which brings the required `intl-imports.js` command:
> 
> * [chore(deps): update frontend-platform version frontend-component-footer#286](https://github.com/openedx/frontend-component-footer/pull/286)
> * [chore(deps): update frontend-platform version frontend-component-header#329](https://github.com/openedx/frontend-component-header/pull/329)
> * [feat: new intl-imports script for atlas src/i18n/index.js generation frontend-platform#463](https://github.com/openedx/frontend-platform/pull/463)

All the linked PRs are merged now!

**Comment by OmarIthawi** [comment]
Thanks @brian-smith-tcril. We still need to tag new releases on both the header and footer on GitHub so they're published on npm.

**Comment by brian-smith-tcril** [comment]
> We still need to tag new releases on both the header and footer on GitHub so they're published on npm.

Releases are out for header and footer now!
https://github.com/openedx/frontend-component-header/releases/tag/v3.7.3
https://github.com/openedx/frontend-component-footer/releases/tag/v11.7.3

**Comment by OmarIthawi** [comment]
> > We still need to tag new releases on both the header and footer on GitHub so they're published on npm.
> 
> Releases are out for header and footer now! https://github.com/openedx/frontend-component-header/releases/tag/v3.7.3 https://github.com/openedx/frontend-component-footer/releases/tag/v11.7.3

Thanks! It turns out we have another blocker. I've updated the PR description: https://github.com/openedx/frontend-lib-special-exams is stuck on `frontend-platform@3.x.x` and needs the following PR to be merged:

- https://github.com/openedx/frontend-lib-special-exams/pull/89

**Comment by OmarIthawi** [comment]
@mphilbrick211 this is now ready for review.

**Comment by mphilbrick211** [comment]
> @mphilbrick211 this is now ready for review.

@itsjeyd FYI!

**Review by brian-smith-tcril — APPROVED** [review]
LGTM! Once conflicts are resolved let's :shipit:

**Comment by itsjeyd** [comment]
Thanks for the ping @mphilbrick211 :)

**Comment by OmarIthawi** [comment]
Conflicts are fixed again. Thanks all for the reviews and help!!

**Comment by brian-smith-tcril** [comment]
@itsjeyd conflicts are resolved on this one!

**Comment by itsjeyd** [comment]
@OmarIthawi @brian-smith-tcril Thanks for the update! Marking this as ready for merge.

@brian-smith-tcril If you have permission to merge (?) please go ahead, otherwise we can ping the Aurora team; let me know.

**Comment by openedx-webhooks** [comment]
@OmarIthawi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-learning&entry.867055334=https://github.com/openedx/frontend-app-learning/pull/1093&entry.1484655318=https://github.com/OmarIthawi&entry.752974735=2023-03-30+16:23&entry.1917517419=2023-05-09+14:03&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**